src/Compilers/CSharp/Test/Emit/PrivateProtected.cs

```diff
@@ -452,5 +576,218 @@ private protected interface I {}                       // 8
+                    Signature("Program", "F", ".field famandassem instance System.Int32 F"),
+                });
+        }
+
+        [Fact]
+        public void VerifyPartialPartsMatch()
+        {
+            var source =
+@"class Outer
+{
+    private protected partial class Inner {}
+    private           partial class Inner {}
+}";
+            CreateStandardCompilation(source, parseOptions: TestOptions.Regular7_2)
+                .VerifyDiagnostics(
+                // (3,37): error CS0262: Partial declarations of 'Outer.Inner' have conflicting accessibility modifiers
+                //     private protected partial class Inner {}
+                Diagnostic(ErrorCode.ERR_PartialModifierConflict, "Inner").WithArguments("Outer.Inner").WithLocation(3, 37)
+                );
+            source =
+@"class Outer
+{
+    private protected partial class Inner {}
+    private protected partial class Inner {}
+}";
+            CreateStandardCompilation(source, parseOptions: TestOptions.Regular7_2)
+                .VerifyDiagnostics(
+                );
+        }
+
+        [Fact]
+        public void VerifyProtectedSemantics()
+        {
+            var source =
+@"class Base
+{
+    private protected void M()
+    {
+        System.Console.WriteLine(this.GetType().Name);
+    }
+}
+
+class Derived : Base
+{
+    public void Main()
+    {
+        Derived derived = new Derived();
+        derived.M();
+        Base bb = new Base();
+        bb.M(); // error 1
+        Other other = new Other();
+        other.M(); // error 2
+    }
+}
+
+class Other : Base
+{
+}";
+            CreateStandardCompilation(source, parseOptions: TestOptions.Regular7_2)
+                .VerifyDiagnostics(
+                // (16,12): error CS1540: Cannot access protected member 'Base.M()' via a qualifier of type 'Base'; the qualifier must be of type 'Derived' (or derived from it)
+                //         bb.M(); // error 1
+                Diagnostic(ErrorCode.ERR_BadProtectedAccess, "M").WithArguments("Base.M()", "Base", "Derived").WithLocation(16, 12),
+                // (18,15): error CS1540: Cannot access protected member 'Base.M()' via a qualifier of type 'Other'; the qualifier must be of type 'Derived' (or derived from it)
+                //         other.M(); // error 2
+                Diagnostic(ErrorCode.ERR_BadProtectedAccess, "M").WithArguments("Base.M()", "Other", "Derived").WithLocation(18, 15)
+                );
+        }
+
+        [Fact]
+        public void HidingAbstract()
+        {
+            var source =
+@"abstract class A
+{
+    internal abstract void F();
+}
+abstract class B : A
+{
+    private protected new void F() { } // No CS0533
+}";
+            CreateStandardCompilation(source, parseOptions: TestOptions.Regular7_2)
+                .VerifyDiagnostics(
+                );
+        }
+
+        [Fact]
+        public void HidingInaccessible()
```

Comment:
Should the test be called `HidingAccessible` instead?

Reply:
It is not accessible because it is in a different assembly.

---
In reply to: [137604830](https://github.com/dotnet/roslyn/pull/21742#discussion_r137604830) [](ancestors = 137604830)